control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1352,6 +1355,51 @@ func (r *HostedControlPlaneReconciler) reconcileCloudProviderConfig(ctx context.
 		}); err != nil {
 			return fmt.Errorf("failed to reconcile Azure cloud config with credentials: %w", err)
 		}
+	case hyperv1.KubevirtPlatform:
```

Comment:
Can we follow the existing structure rather than putting this into `reconcielCloudProviderConfig` https://github.com/openshift/hypershift/pull/1422/files#diff-3d4ffda0834e09a59a4d510523932e20b9234bcb9816e40258a0fb32a87a7006 ?

Reply:
Can you please elaborate here?
Please notice the resources created here (ServiceAccount, Role, RoleBinding, ClusterRole and ClusterRoleBinding) are special case for KubevirtPlatform
The Service Account created here is used to create the credentials (KubeConfig) for the specific cloud-config of the kubevirt platform

Reply:
I moved all the code of resources for the cloudconfig creation (ServiceAccount, Role, RoleBinding, ClusterRole and ClusterRoleBinding) into `cloud/ccm/kubevirt/credentials.go` in order to avoid confusions

Reply:
Right, but you are creating it in the mgmt cluster which is wrong, you need to create it in the guest cluster using the resources controller of the hostedclusterconfigoperator.

The SA doesn't need to be created, you can just reference it from a certificate kubeconfig, it doesn't need to actually exist: https://github.com/openshift/hypershift/blob/c5900c3a93a5c9199eff91f9e503bc0e53f28c9f/control-plane-operator/controllers/hostedcontrolplane/pki/kas.go#L70